tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -804,6 +804,15 @@ GDScript:
           in the shader. It must match *exactly* to the name of the uniform in
           the shader or else it will not be recognized.
 
+.. note:: There is a limit to the total size of shader uniforms that you can use
+          in a single shader. On most desktop platforms, this limit is ``65536``
+          bytes, or ``4096`` ``vec4`` uniforms. On mobile platforms, the limit is smaller.
+          Uniforms smaller than a ``vec4``, such as ``float`` or ``vec2``, are
+          padded to the size of a ``vec4``. Arrays count as the total size of their
```

Comment:
```suggestion
          Vector uniforms smaller than a ``vec4``, such as ``vec2`` or ``vec3``, are
          padded to the size of a ``vec4``. Arrays count as the total size of their
```

Floats won't be padded, nor will ints or other scalar values. Bools are padded to the size of ints